src/Autoloader.php

```diff
@@ -116,17 +116,14 @@ public static function getPsr4()
     /**
      *
      */
-    public static function addClassMap(array $classMap)
+    public static function addClassMap(array $classMap, $path)
     {
         // If not registered, register.
         if (!static::isRegistered()) {
             static::register();
         }
-
-        static::$classMap = array_merge(
-            static::$classMap,
-            $classMap
-        );
+        $path = rtrim($path, DIRECTORY_SEPARATOR);
+        static::$classMap[$path] = $classMap;
```

Comment:
If multiple class maps use the same path (`/usr/share/php` for example), this would overwrite any previous maps.  I think we should take that into account and do an `array_merge` if the class map key (i.e. `$path`) already exists.